src/main/webapp/script.js

```diff
@@ -29,40 +35,93 @@ function createTable() { // eslint-disable-line no-unused-vars
         currentRow.appendChild(keyEntry);
         console.log('The key is ' + key);
         // Create row goes here
-        initializeRow(json, key, currentRow);
+        appendRow(runInformation, key, currentRow);
         table.appendChild(currentRow);
         even = !even;
       }
     }
   });
 }
+
+
 /**
  * Create a row in the given table.
- * @param {Object} json The JSON that is currently being parsed.
+ * @param {Object} runInformation  that is currently being parsed.
  * @param {String} key The key to lookup in the JSON element.
  * @param {Object} currentRow The currentRow to append the new elements too.
  * @return {Element}
  */
-function initializeRow(json, key, currentRow) {
+function appendRow(runInformation, key, currentRow) {
   let index = 0;
-  for (const type of json[key]) {
+  for (const type of runInformation[key]) {
     const columnEntry = document.createElement('td');
-    console.log(type);
-    console.log(type.toUpperCase());
+
     if ((type.toUpperCase() === 'POSE' && index === 0) ||
       (type.toUpperCase() === 'POINTCLOUD' && index === 1)) {
       const link = document.createElement('a');
-      link.href = '/getrun?id=' + key + '&dataType=' + json[key];
+      link.href = '/getrun?id=' + key + '&dataType=' + runInformation[key];
       link.innerText = type;
       columnEntry.appendChild(link);
+      currentRow.appendChild(columnEntry);
+      /* Makes checkboxes for each data entry. */
+      makeCheckbox(currentRow, index, link);
     } else {
       columnEntry.innerText = '';
+      currentRow.appendChild(columnEntry);
     }
-    currentRow.appendChild(columnEntry);
     index++;
   }
   return currentRow;
 }
+
+/**
+* Makes checkboxes for each data entry.
+* @param {object} currentRow currentRow to add checkbox to
+* @param {number} index row checkbox will be on
+* @param {link} link associated with row checkbox will be on
+*/
+function makeCheckbox(currentRow, index, link) {
+  const checkbox = document.createElement('INPUT');
+  checkbox.type = 'checkbox';
+  checkbox.setAttribute('name', 'check');
+  checkbox.setAttribute('value', link.getAttribute('name'));
+  currentRow.appendChild(checkbox);
+  checkArray.push(checkbox);
+}
+
+/* eslint-disable */
+/**
+* Select all Checkboxes function.
+ * @param {object} source checkbox
+ */
+function toggle(source) {
```

Comment:
"toggle" suggests it will invert the state of the boxes, but I think this actually sets them to match the input box?

Reply:
in coding examples i found online for checking checkboxes, they used the name toggle to describe the same action as what i do in this function: 
https://www.developphp.com/video/JavaScript/Checkbox-Toggle-Tutorial-Select-Deselect-All-Elements
https://stackoverflow.com/questions/37286629/toggle-checkbox-using-javascript

This is the first time I'm learning of the word, so i wanted to use it and mirror how other people used it in practice, but can change name to checkAllBoxes